lib/async/async-hook.js

```diff
@@ -71,37 +53,44 @@ const hook = asyncHooks.createHook({
 
     const executionAsyncId = asyncHooks.executionAsyncId();
 
-    const executionChanged = trigger?.executionAsyncId !== executionAsyncId;
-
     const res = {
       asyncId,
       triggerAsyncId,
       executionAsyncId,
-      type,
       start: Ntp._now(),
       end: null,
+      startTime: Ntp._now(),
+      endTime: null,
       activeEvent,
-      executionChanged,
     };
 
     if (Kadira.options.enableAsyncStackTraces || process.env.ENABLE_ASYNC_STACK_TRACES) {
       res.stack = stackTrace();
     }
 
+    res.event = Kadira.tracer.event(info.trace, EventType.Async, pick(res, ['asyncId', 'stack', 'asyncId', 'triggerAsyncId', 'executionAsyncId']));
     info.resources.set(asyncId, res);
```

Comment:
With the async events, we should be able to mostly remove the info.resources and instead use the async events for the async/compute metrics.